File: doc/tutoNLP/positions_spans_marks/l5_sequence_span_loop_body.ml
open Optitrust
open Prelude

(* Difficulty: Level 5
   Request: target the whole sequence inside the bi loop
   Expected target: [tSpanSeq [cForBody "bi"]]
   Accepted variant: [cForBody "bi"; tBetweenAll] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [tSpanSeq [cForBody "bi"]];
)

File: doc/tutoNLP/positions_spans_marks/l5_sequence_span_loop_body.cpp
// Difficulty: Level 5
// Request: target the whole sequence inside the bi loop

void dot(float* a, float* b, float* out) {
  for (int bi = 0; bi < 128; bi++) {
    float t = 0.0f;
    t += a[bi] * b[bi];
    out[bi] = t;
  }
}
